src/instructlab/model/backends/vllm.py

```diff
@@ -15,26 +18,56 @@
 # Local
 from ...configuration import get_api_base
 from .backends import (
+    CHAT_TEMPLATE_AUTO,
+    CHAT_TEMPLATE_TOKENIZER,
     VLLM,
     BackendServer,
     ServerException,
     ensure_server,
+    get_model_template,
     shutdown_process,
 )
 
 logger = logging.getLogger(__name__)
 
 
+class Closeable(typing.Protocol):
+    def close(self) -> None: ...
+
+
+# Temporary global resource cleanup until a better life-cycle contract is introduced
+backend_resources: list[Closeable] = []
+
+
+def register_backend_resources(resources: typing.Iterable[Closeable]) -> None:
+    backend_resources.extend(resources)
+
+
+def cleanup_backend() -> None:
+    safe_close_all(backend_resources)
+
+
+def safe_close_all(resources: typing.Iterable[Closeable]) -> None:
+    for resource in resources:
+        with contextlib.suppress(Exception):
+            resource.close()
+
+
+atexit.register(cleanup_backend)
+
+
```

Comment:
How is this related to `chat_template`?

Reply:
The Auto handler requires the use of temp files that we need to cleanup, yet still stick around long enough for vLLM to open and read them (see the first bullet in the notes section for further details)

Reply:
Ahhhh ok ok :) Thanks for the clarification.